Author: Tomasz Karczewski <tomasz.2173644@infosys.com>
Source: includes simpleshell protocol definition (https://github.com/rdkcentral/westeros/blob/develop/simpleshell/protocol/simpleshell.xml)
Date: Thu, 14 May 2026 13:51:43 +0200

Adding support for RDK-specific simpleshell protocol.
---
 cmake/wayland.cmake                      |   5 +
 cmake/waypp.h.in                         |  12 +-
 third_party/rdk/protocol/simpleshell.xml | 189 +++++++++++++++++++++++
 3 files changed, 205 insertions(+), 1 deletion(-)
 create mode 100644 third_party/rdk/protocol/simpleshell.xml

diff --git a/cmake/wayland.cmake b/cmake/wayland.cmake
index c8c2526..7c1db93 100644
--- a/cmake/wayland.cmake
+++ b/cmake/wayland.cmake
@@ -22,6 +22,7 @@ option(ENABLE_XDG_CLIENT "Enable XDG Client" ON)
 option(ENABLE_AGL_SHELL_CLIENT "Enable AGL shell Client" ON)
 option(ENABLE_IVI_SHELL_CLIENT "Enable ivi-shell Client" OFF)
 option(ENABLE_DRM_LEASE_CLIENT "Enable DRM Lease Client" OFF)
+option(ENABLE_SIMPLE_SHELL_CLIENT "Enable RDK simple_shell Client" OFF)
 
 find_package(PkgConfig REQUIRED)
 pkg_check_modules(WAYLAND REQUIRED IMPORTED_TARGET wayland-client wayland-cursor xkbcommon)
@@ -107,6 +108,10 @@ if (ENABLE_IVI_SHELL_CLIENT)
     message(STATUS "IVI WM ................ ${HAS_WAYLAND_PROTOCOL_IVI_WM}")
 endif ()
 
+if (ENABLE_SIMPLE_SHELL_CLIENT)
+    add_protocol(${PROJECT_SOURCE_DIR}/third_party/rdk/protocol/simpleshell.xml)
+endif ()
+
 add_protocol(${PROJECT_SOURCE_DIR}/third_party/weston/protocol/weston-output-capture.xml)
 
 
diff --git a/cmake/waypp.h.in b/cmake/waypp.h.in
index 0862ece..a0a0dba 100644
--- a/cmake/waypp.h.in
+++ b/cmake/waypp.h.in
@@ -36,6 +36,9 @@
 #ifndef ENABLE_IVI_SHELL_CLIENT
 #cmakedefine01 ENABLE_IVI_SHELL_CLIENT
 #endif
+#ifndef ENABLE_SIMPLE_SHELL_CLIENT
+#cmakedefine01 ENABLE_SIMPLE_SHELL_CLIENT
+#endif
 #ifndef ENABLE_DRM_LEASE_CLIENT
 #cmakedefine01 ENABLE_DRM_LEASE_CLIENT
 #endif
@@ -106,6 +109,9 @@
 #ifndef HAS_WAYLAND_PROTOCOL_PRIMARY_SELECTION_UNSTABLE_V1
 #cmakedefine01 HAS_WAYLAND_PROTOCOL_PRIMARY_SELECTION_UNSTABLE_V1
 #endif
+#ifndef HAS_WAYLAND_PROTOCOL_SIMPLE_SHELL
+#cmakedefine01 HAS_WAYLAND_PROTOCOL_SIMPLE_SHELL
+#endif
 
 #ifndef HAVE_MEMFD_CREATE
 #cmakedefine01 HAVE_MEMFD_CREATE
@@ -136,6 +142,10 @@
 #include "ivi-wm-client-protocol.h"
 #endif
 
+#if ENABLE_SIMPLE_SHELL_CLIENT
+#include "simpleshell-client-protocol.h"
+#endif
+
 #if HAS_WAYLAND_PROTOCOL_DRM_LEASE_V1
 #include "drm-lease-v1-client-protocol.h"
 #endif
@@ -189,4 +199,4 @@
 #include "primary-selection-unstable-v1-client-protocol.h"
 #endif
 
-#endif  // INCLUDE_WAYPP_WAYPP_H_
\ No newline at end of file
+#endif  // INCLUDE_WAYPP_WAYPP_H_
diff --git a/third_party/rdk/protocol/simpleshell.xml b/third_party/rdk/protocol/simpleshell.xml
new file mode 100644
index 0000000..304e27d
--- /dev/null
+++ b/third_party/rdk/protocol/simpleshell.xml
@@ -0,0 +1,189 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<protocol name="simple_shell">
+
+    <copyright>
+    If not stated otherwise in this file or this component's LICENSE file the
+    following copyright and licenses apply:
+
+    Copyright 2016 RDK Management
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+    -------
+    Copyright © 2008-2011 Kristian Høgsberg
+    Copyright © 2010-2011 Intel Corporation
+    Permission to use, copy, modify, distribute, and sell this
+    software and its documentation for any purpose is hereby granted
+    without fee, provided that\n the above copyright notice appear in
+    all copies and that both that copyright notice and this permission
+    notice appear in supporting documentation, and that the name of
+    the copyright holders not be used in advertising or publicity
+    pertaining to distribution of the software without specific,
+    written prior permission.  The copyright holders make no
+    representations about the suitability of this software for any
+    purpose.  It is provided "as is" without express or implied
+    warranty.
+    THE COPYRIGHT HOLDERS DISCLAIM ALL WARRANTIES WITH REGARD TO THIS
+    SOFTWARE, INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY AND
+    FITNESS, IN NO EVENT SHALL THE COPYRIGHT HOLDERS BE LIABLE FOR ANY
+    SPECIAL, INDIRECT OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN
+    AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,
+    ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF
+    THIS SOFTWARE.
+  </copyright>
+  <interface name="wl_simple_shell" version="1">
+
+    <description summary="control the layout of surfaces">
+      The simple_shell provides control over the size, position,
+      z-order, opacity, etc., of the surfaces making up the user
+      interfaces suitable for situations such as embedded systems.
+    </description>
+
+    <!-- Raised when a new surface is created to provide the surfaceId -->
+    <event name="surface_id">
+      <description summary="provides surface id for new surface">
+   This event is sent to the creator of a new wl_surface to
+   provide the surface id.  The surface id is used in the
+   simple_shell protocol to control the surface.  The surface
+   creator can set a surface name with the set_name request
+   in response to this event.
+      </description>
+      <arg name="surface" type="object" interface="wl_surface"/>
+      <arg name="surfaceId" type="uint"/>
+    </event>
+    <!-- Raised when a new surface is created by any client-->
+    <event name="surface_created">
+      <description summary="broadcast announcement of a new surface">
+   This event is sent to all simple_shell listeners to announce the creation
+   of a new wl_surface.
+      </description>
+      <arg name="surfaceId" type="uint"/>
+      <arg name="name" type="string"/>
+    </event>
+    <!-- Raised when a surface is destroyed by any client-->
+    <event name="surface_destroyed">
+      <description summary="broadcast announcement of a surface destruction">
+   This event is sent to all simple_shell listeners to announce the destruction
+   of a new wl_surface.
+      </description>
+      <arg name="surfaceId" type="uint"/>
+      <arg name="name" type="string"/>
+    </event>
+    <!-- Raised in response to a get_status request for a surface -->
+    <event name="surface_status">
+      <description summary="supply current surface information">
+   This event is sent in response to a get_status request to the listener
+   registered by the client issuing the request.
+      </description>
+      <arg name="surfaceId" type="uint"/>
+      <arg name="name" type="string"/>
+      <arg name="visible" type="uint"/>
+      <arg name="x" type="int"/>
+      <arg name="y" type="int"/>
+      <arg name="width" type="int"/>
+      <arg name="height" type="int"/>
+      <arg name="opacity" type="fixed"/>
+      <arg name="zorder" type="fixed"/>
+    </event>
+    <!-- Raised to mark the end of surface_status events for get_surfaces -->
+    <event name="get_surfaces_done">
+      <description summary="marks end of surface_status events for get_surfaces">
+   This event is sent to a client that issues a get_surfaces request after the last
+   surface_status event is sent.
+      </description>
+    </event>
+
+    <event name="popup_details">
+      <description summary="sent for the response of is_surface_popup">
+      </description>
+      <arg name="surfaceId" type="uint"/>
+      <arg name="parentSurfaceId" type="uint" summary="this will be 0 if the popup does not have a parent or the surface is not a popup"/>
+      <arg name="popup" type="int" summary="boolean for whether surface is a popup or not"/>
+    </event>
+
+    <!-- Set the name of a surface -->
+    <request name="set_name">
+      <arg name="surfaceId" type="uint"/>
+      <arg name="name" type="string"/>
+    </request>
+
+    <!-- Set the visibility of a surface -->
+    <request name="set_visible">
+      <arg name="surfaceId" type="uint"/>
+      <arg name="visible" type="uint"/>
+    </request>
+
+    <!-- Set the geometry of a surface -->
+    <request name="set_geometry">
+      <arg name="surfaceId" type="uint"/>
+      <arg name="x" type="int"/>
+      <arg name="y" type="int"/>
+      <arg name="width" type="int"/>
+      <arg name="height" type="int"/>
+    </request>
+
+    <!-- Set the opacity of a surface -->
+    <request name="set_opacity">
+      <arg name="surfaceId" type="uint"/>
+      <arg name="opacity" type="fixed"/>
+    </request>
+
+    <!-- Set the z-order of a surface -->
+    <request name="set_zorder">
+      <arg name="surfaceId" type="uint"/>
+      <arg name="zorder" type="fixed"/>
+    </request>
+
+    <!-- Query the status of a surface -->
+    <request name="get_status">
+      <arg name="surfaceId" type="uint"/>
+    </request>
+
+    <!-- Query all existing surfaces -->
+    <request name="get_surfaces">
+      <description summary="get all existing surfaces">
+      Causes a surface status event for each existing surface
+      to be sent to the listener registered by the client issuing
+       the request.
+      </description>
+    </request>
+
+    <!-- Set the focus on a surface -->
+    <request name="set_focus">
+      <arg name="surfaceId" type="uint"/>
+    </request>
+
+    <!-- Set the scale on a surface -->
+    <request name="set_scale">
+      <arg name="surfaceId" type="uint"/>
+      <arg name="scaleX" type="fixed"/>
+      <arg name="scaleY" type="fixed"/>
+    </request>
+
+    <request name="get_popup">
+      <description summary="creates a popup surface that is anchored to the specified parent"/>
+      <arg name="surfaceId" type="uint"/>
+      <arg name="parentSurfaceId" type="uint"/>
+      <arg name="x" type="int"/>
+      <arg name="y" type="int"/>
+      <arg name="width" type="int"/>
+      <arg name="height" type="int"/>
+    </request>
+
+    <request name="is_surface_popup">
+      <arg name="surfaceId" type="uint"/>
+    </request>
+
+  </interface>
+
+</protocol>
-- 
2.43.0

